### PR TITLE
Add test for help function

### PR DESCRIPTION
**Pull request by jemrobinson**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).
- [x] You have marked this pull request as a **draft** and added `'[WIP]'` to the title if needed (if you're not yet ready to merge).

### :closed_umbrella: Related issues

Addresses point from #1853 review

### :microscope: Tests

Tested locally

### CODE REVIEW

File: data_safe_haven/cli.py [hunk 1]
```diff
@@ -90,9 +91,13 @@ def main() -> None:
...
 
-    # Start the application
+    return application
+
+
+def main() -> None:
+    """Run the application and log any exceptions"""
     try:
-        application()
+        application()()
```
Comment (JimMadge) [inline data_safe_haven/cli.py]: Is there a reason to have this as a factory rather than a function?

**Review by JimMadge — CHANGES_REQUESTED** [review]

File: data_safe_haven/cli.py (hunk 1 above)
Comment (jemrobinson) [reply data_safe_haven/cli.py]: If you want to make this change then you'll need to change the return type of `application` from `typer.Typer` to whatever a `typer.Typer()` is.

**Review by JimMadge — CHANGES_REQUESTED** [review]
In general, I don't like how `cli.py` is structured. It isn't very idiomatic typer.

I think it would be much clearer and easier to use if it followed the normal model of a `typer.Typer` object and `typer.Typer.command` decorators.

**Comment by jemrobinson** [comment]
> In general, I don't like how `cli.py` is structured. It isn't very idiomatic typer.
> 
> I think it would be much clearer and easier to use if it followed the normal model of a `typer.Typer` object and `typer.Typer.command` decorators.

Are you happy to make that change on this PR?

**Review by JimMadge — APPROVED** [review]

**Pull request merged**